File: docs/article/source/csharp/apps/cargo-tracker/src/CargoTracker.Web/Tracking/Domain/Events/TrackingExceptionResolvedEvent.cs
using MediatR;

namespace CargoTracker.Tracking.Domain.Events;

/// <summary>
/// 追跡例外が解決されたことを表すドメインイベント（US19/US20 の対応報告）。
/// 荷主への対応報告通知（記録で代替）に用いる（ADR-0009 の結果整合性方針に従う）。
/// BC 独立のためプリミティブのみを運ぶ。
/// </summary>
public sealed record TrackingExceptionResolvedEvent(
    string BookingId,
    string TrackingNumber,
    string ExceptionType,
    DateTimeOffset ResolvedAt,
    string? ResolutionNotes) : INotification;

File: docs/article/source/java-4/apps/cargo-tracker/backend/trackingms/src/main/java/com/example/cargotracker/tracking/domain/model/events/TrackingExceptionResolvedEvent.java
package com.example.cargotracker.tracking.domain.model.events;

import java.time.Instant;
import org.axonframework.eventsourcing.annotation.EventTag;

/**
 * 例外が解決した（UC16 / US19 §受入基準 4）。
 *
 * <p><b>例外そのものは消えない</b>（不変条件 6）。消えるのは「対応が要る」という
 * 状態だけで、起きた事実は料金調整の根拠として残る。</p>
 */
public record TrackingExceptionResolvedEvent(
        @EventTag(key = "trackingNumber") String trackingNumber,
        String exceptionId,
        String resolution,
        String resolvedBy,
        Instant resolvedAt) {
}
